### ISSUE TITLE
script file is not working by adding on angular.jason file inside script array

### ISSUE THREAD

**Issue by altaf596** [stackblitz/core#700]
When we are going to refrence any script file from angular project folder or from node_modules package folder it is not working. 

Ex: Suppose we are going to add bootstrap package to the angular project.  Then we are getting 4 important file that must to add as reference file.
1. Jquery
2. popper.js
3. bootstrap.css
4. boostrap.js

On the above package boostrap.css file is working good. But script files are not working.
Files are added as the bellow formating:

Inside angular.json file:
 "styles": [
              "src/styles.css",
              "node_modules/bootstrap/dist/css/bootstap.min.css"
            ],
"scripts": [
"node_modules/jquery/dist/jquery.min.js",
"node_modules/popper.js/dist/umd/popper.min.js",
"node_modules/bootstrap/dist/js/bootstap.min.js"
]

According the the local system It should to work but script file is not working.

We are waiting to have fixed this issue soon.

Thanks so much to Visual Studio Code team.

**Comment by stephanrauh** [comment]
Same here. I've got a reproducer for you: https://stackblitz.com/edit/angular-jxny3v

The two files in the `scripts` section of the Angular.json are ignored:

```json
"scripts": [
              "node_modules/ngx-extended-pdf-viewer/assets/pdf-es5.js",
              "node_modules/ngx-extended-pdf-viewer/assets/viewer-es5.js"
            ]
```

**Comment by lula** [comment]
[https://stackblitz.com/edit/angular-summernote-demo-zwzcyj](https://stackblitz.com/edit/angular-summernote-demo-zwzcyj)

JQuery is not found:

<img width="428" alt="Screenshot 2020-02-12 at 01 20 37" src="https://user-images.githubusercontent.com/614544/74291661-e4799480-4d35-11ea-924e-9c5135dabec7.png">

I guess styles might not be found as well

**Comment by dev-sharan** [comment]
![image](https://user-images.githubusercontent.com/18690573/74434816-367eff00-4e89-11ea-9ae4-6ed88a93d424.png)

**Comment by lula** [comment]
thanks! but it doesn't seem to work 😕

**Comment by stephanrauh** [comment]
@lula Actually, your clever solution does work on my machine. Maybe it was a temporary error?

**Comment by dev-sharan** [comment]
> thanks! but it doesn't seem to work

https://stackblitz.com/edit/angular-mj2wef

**Comment by lula** [comment]
@stephanrauh I gave no solution so far 😄 

Still, @dev-sharan workaround it's not working to me:

<img width="1278" alt="Screenshot 2020-02-13 at 22 05 11" src="https://user-images.githubusercontent.com/614544/74478428-a1dac800-4ead-11ea-8c0e-a7b1f6334511.png">

Tested with Chrome Version 79.0.3945.130 (Official Build) (64-bit) and with Safari Version 13.0.4 (15608.4.9.1.3) on macOS X Catalina 10.15.2.

Besides, what's the point on having angular.json if it's not taken into consideration? A workaround could be fine, but I still believe the issue should at least be clarified somehow.

**Comment by dev-sharan** [comment]
@lula sure thing

**Comment by stephanrauh** [comment]
I've followed the link given by @dev-sharan. It looks like so on my PC:
![image](https://user-images.githubusercontent.com/3045767/74480192-c1272480-4eb0-11ea-8244-7afe670cde53.png)

**Comment by stephanrauh** [comment]
@lula As for the Angular.json support: 100% agreed. That's why I chimed in into this discussion in the first place. Dear Stackblitz team, what's your opinion on this topic? 

Background information: Many users of my open-source project (ngx-extended-pdf-viewer) would love to send me bug reports using a Stackblitz demo. The missing angular.json support makes this difficult. FWIW, I'd like to provide demos using Stackblitz, too.

**Comment by lula** [comment]
@stephanrauh yeah I always asked to create testable report on stackblitz for issues on my lib as well 😆 

@dev-sharan sorry, I haven't seen you message before. 

Anyway, I can actually get $ reference in the console. Then there must be somethings that prevents my lib to work in stackblitz.

**Comment by stephanrauh** [comment]
Maybe a timing problem? Do you call jQuery too early?

**Comment by lula** [comment]
I don't know. It seems to happen [here](https://github.com/lula/ngx-summernote/blob/e0fd9a4c978beeca334f0206616f0e4c4c07d2d7/projects/ngx-summernote/src/lib/ngx-summernote.directive.ts#L249). Method is called in ngOnInit. I never had a similar issue running a project with this lib (I use it myself in my work projects), except when I forget to add jquery 😄

**Comment by stephanrauh** [comment]
Ah, I see. Until now, I've only seen a small fraction of your code. Bear with me.

From what I've gathered, you're loading jQuery lazily in your `main` class. Or something like that. In any case, you insert a DOM node into the source code that causes jQuery to be loaded.

That's a good workaround to the base problem. For some reason, Stackblitz mostly ignores the `angular.json`, so adding a `<script>` node is probably the only way to load jQuery.

Thing is, loading jQuery takes some time. In the meantime, your `ngOnInit()` method is executed.

Most likely everything works if you add a timeout to your jQuery call. Depending on your internet connection, one second should be more than enough. Maybe more seconds if you're using a cell phone outside the big cities. My pet project (ngx-extended-pdf-viewer) loops until the library is loaded.

I don't think you need this, but just in case, here's the link to my sourcecode: https://github.com/stephanrauh/ngx-extended-pdf-viewer/blob/4dbf5439018a4a9253dc217e00f93685e32be3df/projects/ngx-extended-pdf-viewer/src/lib/ngx-extended-pdf-viewer.component.ts#L456 

The library I'm loading adds a property to the `windows` scope, so I wait until this property has been set before procedding.

**Comment by lula** [comment]
@stephanrauh thanks for having a look at my code!!! I see yours and make completely sense. Most likely it's a timing issue, jquery might be not loaded at the moment I try get summernote reference. I can do something like you did, which is a clever solution! I'm still doubtful though as in "normal" angular projects everything works just and I'm not really keen on changing something that is supposed to work.

**Comment by stephanrauh** [comment]
@lula Save it as a copy. If you want to play it safe, take the version @dev-sharan sent you (https://stackblitz.com/edit/angular-mj2wef). It's already a copy. AFAIK you get another copy automatically when you save the copy of @dev-sharan .

And you're right, it's a workaround you don't need in a regular Angular project. However, you guys taught me how to get my demos up and running on Stackblitz, so I also have to thank you!

Best regards,
Stephan

**Comment by lula** [comment]
@stephanrauh sure, jquery it's loaded in @dev-sharan version, but the project he shared is different than the first solution he gave me with the addScript function, which is by the way working

In his Angular 6 project, how could he add these: 

<img width="370" alt="Screenshot 2020-02-15 at 17 16 57" src="https://user-images.githubusercontent.com/614544/74591302-1d0bbd80-5017-11ea-9c6c-abdb456171ee.png">

I can't find them anywhere.

Anyway, thanks to your suggestions and @dev-sharans  I was able to make my project work in stackblitz! 🎉 🎉 🎉

**Comment by ayyash** [comment]
Why has this gone stale? does it have a solution we are not aware of?

**Comment by rizkiprasetya** [comment]
Try to load your script file from inside **src** folder.
Something like **src/your_script.js** or **src/your_folder/your_script.js**
